### PR TITLE
Move nextURL from RouterActionCreator to RouterStore

### PR DESCRIPTION
**Pull request by misstick**
## tasks
 - [x] wait for merge of https://github.com/cozy/cozy-emails/pull/898,
 - [x] move code from actions to store,

## FIX
 - [x] show conversation issue: a click on a messageLisItem trigger JavaScript error and no conversations was displayed,
 - [x] messages list issue: `unreadMailbox` or `flaggedMailbox` didnt have any messages when the should have some,

## Tests on methods 
 - [x] getMessagesPerPage
 - [x] hasNextPage
 - [x] getLastPage
 - [x] isPageComplete

### CODE REVIEW

File: client/app/stores/router_store.coffee [hunk 1]
```diff
@@ -53,7 +53,7 @@ class RouterStore extends Store
 
     # Paginate Messages.list
     _pages = {}
-    _nextURL = {}
+    _requests = {}
```
Comment (misstick) [inline client/app/stores/router_store.coffee]: @aenario, @gregorylegarec, @m4dz : you can see that I rename variables to make obvious the relation between these and requestStore

**Comment by misstick** [comment]
@aenario : I didn't have the time to handle these tests:
        - At first nothing should be stored
        - Each request should be saved with a uniq URI
        - Nothing should be stored with falsy `params`

But all the most complicated is already done into `Goto last page should set `isComplete` falsy`; feel free to finish this in another PR :]]

**Comment by m4dz** [comment]
Woohooo \o/ !

File: client/app/router.coffee [hunk 2]
```diff
@@ -105,6 +105,11 @@ _dispatch = (payload, filter) ->
     {mailboxID, conversationID} = payload
     filter = _parseQuery filter if filter
 
+    payload.filter = filter
```
Comment (misstick) [inline client/app/router.coffee]: Payload isnt updated before dispatch that's why I moved this above

**Pull request merged**